src/python/pants/backend/java/target_types.py

```diff
@@ -185,9 +185,9 @@ class DeployJar(Target):
         JvmResolveNameField,
     )
     help = (
-        "A `jar` file that contains the compiled source code along with its dependency class "
-        "files, where the compiled class files from all dependency JARs, along with first-party "
-        "class files, exist in a common directory structure."
+        "A `jar` file with first and third-party code bundled for deploys.\n\n"
+        "The JAR will contain compiled first-party code and class files for "
+        "third-party dependencies. All class files exist in a common directory structure."
```

Comment:
Is this accurate? I didn't totally follow the original.

Reply:
"compiled first party code" == "class files for first party code": i.e., both first and third party code ends up as classfiles.

So this would be equivalent (although probably not better):
```python
        "A `jar` file with first and third-party code bundled for deploys.\n\n"
        "The JAR will contain class files for first-party code and for third-party code, "
        "all in a common directory structure."
```